src/span.rs

```diff
@@ -1,38 +1,32 @@
 use std::ops::Range;
 
-// A source code span, used for error reporting.
-#[derive(Clone, Debug, PartialEq)]
-pub enum Span {
-    // Span is unknown - no source information.
-    Unknown,
-    // Byte range.
-    ByteRange(Range<usize>),
-}
-
-impl Default for Span {
-    fn default() -> Self {
-        Self::Unknown
-    }
+/// A source code span, used for error reporting.
+#[derive(Clone, Copy, Debug, PartialEq, Default)]
+pub struct Span {
+    pub start: u32,
+    pub end: u32,
 }
 
 impl Span {
-    pub fn subsume(&mut self, other: &Self) {
-        match *self {
-            Self::Unknown => self.clone_from(other),
-            Self::ByteRange(ref mut self_range) => {
-                if let Self::ByteRange(ref other_range) = *other {
-                    self_range.start = self_range.start.min(other_range.start);
-                    self_range.end = self_range.end.max(other_range.end);
-                }
-            }
-        }
+    pub fn subsume(&mut self, other: Self) {
```

Comment:
I think this is incorrect. If I have an empty span, like the 0..0 returned by `Default`, then the result of `subsume` will always have `start == 0`, but the behavior we want is that the result would be the other span.